app/controllers/mentors_controller.rb

```diff
@@ -11,12 +11,18 @@ def index
 
     # mentors are filtered by the criteria above
     last_selected_coach = params[:mentor][:filter_by_coach_id]
+    last_selected_meeting_day = params[:mentor][:filter_by_meeting_day]
     if params[:mentor][:filter_by_coach_id].to_i > 0
```

Comment:
This is a local variable for storing the last selected meeting day, so as to stay in the form sidebar after the selection. Does if affect somehow the filtering?